sonar-batch/src/main/java/org/sonar/batch/report/MeasuresPublisher.java

```diff
@@ -31,30 +33,59 @@
 import org.sonar.batch.protocol.output.BatchReport;
 import org.sonar.batch.protocol.output.BatchReportWriter;
 import org.sonar.batch.scan.measure.MeasureCache;
+import org.sonar.core.metric.BatchMetrics;
+
+import static com.google.common.collect.Iterables.filter;
+import static com.google.common.collect.Iterables.transform;
+import static com.google.common.collect.Sets.newHashSet;
 
 public class MeasuresPublisher implements ReportPublisherStep {
 
+  private final class IsMetricAllowed implements Predicate<Measure> {
+    private final Set<String> allowedMetricKeys;
+
+    private IsMetricAllowed(Set<String> allowedMetricKeys) {
+      this.allowedMetricKeys = allowedMetricKeys;
+    }
+
+    @Override
+    public boolean apply(Measure input) {
+      return allowedMetricKeys.contains(input.getMetricKey());
+    }
+  }
+
+  private final class MetricToKey implements Function<Metric, String> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)